### PR TITLE
chore: add commit message linting

### PR DESCRIPTION
**Pull request by ncoden**
## Description

Various changes to clean up Husky scripts and add commit linting.

### Changes:
* upgrade Husky to `v1.0-rc.1` and migrate configuration to `.huskyrc.yml`
* add commit message linting with Commitlint before tests
* rename husky scripts according to their action
* add project Commitizen convention ([cz-conventional-changelog](https://github.com/commitizen/cz-conventional-changelog))

## Types of changes

- [ ] Documentation
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing
      functionality to change)

## Checklist (all required):

- [x] I have read and follow the [CONTRIBUTING](CONTRIBUTING.md) document.
- [x] There are no other pull request similar to this one.
- [x] The pull request title is descriptive.
- [x] The template is fully and correctly filled.
- [x] The pull request targets the right branch (`develop` or `support/*`).
- [x] My commits are correctly titled and contain all relevant information.
- [x] My code follows the code style of this project.
- [x] I have updated the documentation accordingly to my changes (if relevant).
- [x] I have added tests to cover my changes (if relevant).
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by ncoden** [comment]
@DanielRuf Could you review this when you have some time ?

File: script/husky-commit-lint.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+const chalk = require('chalk')
+const spawn = require('child_process').spawn
+
+const args = process.argv.splice(process.execArgv.length + 2);
```
Comment (DanielRuf) [inline script/husky-commit-lint.js]: Some semicolon at the end?

File: script/husky-commit-lint.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+child.on('error', function (err) {
+  console.log(chalk.red(err))
+})
+
+child.on('exit', function (code) {
+  if(code !== 0){
+    console.log(chalk.yellow('🐶  ✗ Commit message is invalid.'))
+    console.log(chalk.yellow('     See https://git.io/contribute for help'))
+    process.exit(code);
```
Comment (DanielRuf) [inline script/husky-commit-lint.js]: Some semicolon at the end?

Comment (DanielRuf) [reply script/husky-commit-lint.js]: Same in the "old" husky scripts.

**Review by DanielRuf — APPROVED** [review]
So far so good, just some code style things.

File: script/husky-commit-lint.js (hunk 1 above)
Comment (ncoden) [reply script/husky-commit-lint.js]: I was actually surprised there were no semicolons in the first place. It seems we use the `semi: [0, "never"]` rule (so no semicolons expected but errors are ignored), but they're used almost everywhere in the code.

I would personally go for the "always" options for the reasons raised in the airbnb eslint convention: https://github.com/airbnb/javascript#semicolons

**Comment by ncoden** [comment]
> b924d2f Merge branch 'develop' into chore/commit-messages-linting

Resolved conflicts

File: script/husky-commit-lint.js (hunk 1 above)
Comment (DanielRuf) [reply script/husky-commit-lint.js]: > semi: [0, "never"]

Hm, never saw any linting errors or notices / warnings about this when we have semicolons. Is our linting setup just throwing warnings?

**Comment by DanielRuf** [comment]
> Resolved conflicts in yarn.lock

Hm, it seems there is another conflict with the lockfile again.

File: script/husky-commit-lint.js (hunk 1 above)
Comment (ncoden) [reply script/husky-commit-lint.js]: If I remember `0` means that errors are ignored. `1` is a warning and `2` is an error.

**Comment by ncoden** [comment]
I guess it's good to go like that.

**Pull request merged**